File: backend/src/HelpDeskLite.Domain/Entities/Ticket.cs
using HelpDeskLite.Domain.Enums;

namespace HelpDeskLite.Domain.Entities;

public class Ticket
{
    public Guid Id { get; set; }
    public string TicketNumber { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Guid CategoryId { get; set; }
    public TicketPriority Priority { get; set; }
    public Guid CreatedByUserId { get; set; }
    public string Status { get; set; } = nameof(TicketStatus.New);
    public Guid? AssigneeId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Category Category { get; set; } = null!;
    public User CreatedByUser { get; set; } = null!;
    public User? Assignee { get; set; }
    public ICollection<TicketAttachment> Attachments { get; set; } = [];
    public ICollection<TicketComment> Comments { get; set; } = [];
    public ICollection<TicketStatusHistory> StatusHistory { get; set; } = [];
    public ICollection<TicketAssignmentHistory> AssignmentHistory { get; set; } = [];
}

File: frontend/src/types/ticket.ts
export type TicketPriority = 'Low' | 'Medium' | 'High' | 'Critical';

export type TicketStatus =
  | 'New'
  | 'Assigned'
  | 'InProgress'
  | 'WaitingForUser'
  | 'Resolved'
  | 'Closed';

export interface Category {
  id: string;
  name: string;
  description?: string;
}

export interface TicketAttachment {
  id: string;
  fileName: string;
  contentType: string;
  fileSizeBytes: number;
}

export interface Ticket {
  id: string;
  ticketNumber: string;
  title: string;
  description: string;
  categoryId: string;
  categoryName: string;
  priority: TicketPriority;
  createdByUserId: string;
  status: string;
  assigneeId?: string | null;
  assigneeName?: string | null;
  createdAt: string;
  attachments: TicketAttachment[];
}

export interface TicketDetail extends Ticket {
  createdByName: string;
  updatedAt?: string | null;
  allowedNextStatuses: string[];
}

export interface TicketActivity {
  id: string;
  activityType: string;
  summary: string;
  detail?: string | null;
  actorId?: string | null;
  actorName?: string | null;
  occurredAt: string;
  isInternal: boolean;
}

export interface TicketComment {
  id: string;
  ticketId: string;
  authorId: string;
  authorName: string;
  comment: string;
  isInternal: boolean;
  createdAt: string;
}

export interface AgentUser {
  id: string;
  fullName: string;
  email: string;
}

export interface CreateTicketForm {
  title: string;
  categoryId: string;
  description: string;
  priority: TicketPriority;
  attachments: File[];
}

export interface UpdateTicketStatusRequest {
  status: string;
}

export interface AssignTicketRequest {
  assigneeId: string | null;
}

export interface CreateTicketCommentRequest {
  comment: string;
  isInternal: boolean;
}

export interface KnowledgeBaseSuggestion {
  id: string;
  title: string;
  summary: string;
  url?: string;
  relevanceScore: number;
}
